### ISSUE TITLE
alpha.cuelang.org: in-CI preprocessor disagrees with local preprocessor

### ISSUE THREAD

**Issue by jpluscplusm** [cue-lang/cue#2668]
### What version of CUE are you using (`cue version`)?

<pre>
$ cue version
cue version v0.6.0

go version go1.20.6
      -buildmode exe
       -compiler gc
       -trimpath true
     CGO_ENABLED 0
          GOARCH amd64
            GOOS linux
         GOAMD64 v1
</pre>

### Does this issue reproduce with the latest stable release?

0.6 is latest stable.

### What did you do?

I wrote some docs for alpha, in https://review.gerrithub.io/c/cue-lang/cuelang.org/+/1171469.

Having rebuilt the preprocessor's docker container, I ran the preprocessor both with and without the `--update` flag, and committed the files it had updated-and-verified - both the cache files and the updated source documents.

CI then failed: https://github.com/cue-lang/cuelang.org-trybot/actions/runs/6706361135/job/18222697128#step:34:2364

The failure appears to be the preprocessor suggesting that various txtars embedded in source markdown aren't valid, as their inputs don't generate identical output to what I committed.

To reproduce in CI, trigger the trybot for https://review.gerrithub.io/c/cue-lang/cuelang.org/+/1171469. I've been unable to reproduce locally, using either `_scripts/build.bash` or `_scripts/runPreprocessor.bash execute`.

### What did you expect to see?

A passing build as the in-CI preprocessor was merely validating the local preprocessor's output.

### What did you see instead?

A failing build in CI.

**Comment by myitcv** [comment]
This isn't actually a bug here as far as I can tell.

The issue lies here (taken from `content/docs/tour/types/optional/en.md`):

```
{{{with code "en" "structs"}}}
exec cue eval -c structs.cue
cmp stdout '$ cue eval -c structs.cue'
-- structs.cue --
#a: {
	foo?: int
	bar?: string
	baz?: string
}
b: #a & {
	foo:  3
	baz?: 2 // baz?: _|_
}
-- $ cue eval -c structs.cue --
b: {
    foo: 3
}
{{{end}}}
```

The "filename" for the second file in the txtar archive should not be the command that has been executed. 

As it stands, the preprocessor thinks it is a `.cue` file and so tries to format it as such.

Formatting currently updates input files (but does not write the updated files back to the input txtar) without requiring the `--update` flag (which itself is rather dubious, but not the root cause here), which then causes the diff when the actual command runs.

TL;DR - we shouldn't be using the command as a filename. It's brittle (because it requires updating in two places) and unnecessary. If we need another mode for the preprocessor, then so be it.

**Comment by jpluscplusm** [comment]
> As it stands, the preprocessor thinks it is a `.cue` file and so tries to format it as such.

Oh, I see - `cue fmt` and `cue eval` can both emit semantically equivalent CUE (based on different inputs, in this case), but with different indentation in their output formatting. Got it.

> TL;DR - we shouldn't be using the command as a filename. It's brittle (because it requires updating in two places) and unnecessary. 

It's definitely brittle but, whilst I thought it worked, it felt like the right trade-off to prioritise the reader's visibility of the non-standard command being executed.

I'll update #2667 to stop referencing this form, so as not to distract from the underlying feature request.
